plotting/plot_factory/plot_factory.py

```diff
@@ -0,0 +1,208 @@
+
+    def extract_layout(self):
+        """Extracts plot layout data from plot style meta data
+
+        :return self.meta_data (dictionary)
+        """
+        interpreted_layout = self.read_plot_meta_data()
+
+        if 'mode' in interpreted_layout:
+            self.mode = interpreted_layout.pop('mode')
+        if 'plot' in interpreted_layout:
+            self.plot_type = interpreted_layout.pop('plot')
+        if 'error_bars' in interpreted_layout:
+            self.error_bars = interpreted_layout.pop('error_bars')
+        return interpreted_layout
+
+
+class Trace:
+    """Creates a trace object """
+    def __init__(self, data, plot_style, plot_name, mode=None, error_bars=None): #pylint: disable=too-many-arguments, line-too-long
+        """
+        :param data (pandas dataframe)
+        :param plot_style (dictionary)
+        :param plot_name (string)
+        :param mode (string)
+        :param error_bars (bool)
+        """
+
+        self.trace = self.create_trace(mode=mode,
+                                       data=data,
+                                       plot_style=plot_style,
+                                       name=plot_name,
+                                       error_bars=error_bars)
+
+    @staticmethod
+    def trace_dict(data, error_bars):
+        """Creates trace dictionary
+
+        :param data (dataframe)
+        :param error_bars (bool)
+        """
+        trace = {}
+        for axis in list(data.columns):
+            if axis == 'E':
+                trace['error_y'] = dict(type='data',
+                                        array=data[axis].to_list(),
+                                        visible=error_bars)
+                # else:
+                #     trace[axis.lower()] = f"data['{axis}']"
+            else:
+                trace[axis.lower()] = f"data['{axis}']"
+        return trace
+
+    @staticmethod
+    def dict_to_string(trace_dictionary):
+        """Converts a dictionary ot a string
+        :param trace_dictionary
+
+        :return object
+        """
+
+        return ', '.join([f"{key}= {value}" for key, value in trace_dictionary.items()])
+
+    def create_trace(self, data, plot_style, name, error_bars, mode=None): #pylint: disable=too-many-arguments, line-too-long
+        """Creates a trace
+        :param data (dataframe)
+        :param plot_style (dictionary)
+        :param name (string)
+        :param error_bars (bool)
+        :param mode (string)
+
+        :return figure (object)
+        """
+        # make dictionary with string values
+        trace = self.trace_dict(data=data, error_bars=error_bars)
+        trace['name'] = f"'{name}'"
+
+        if mode:
+            trace['mode'] = f"'{mode}'"
+
+        # make dictionary string
+        trace_as_string = self.dict_to_string(trace)
+
+        # perform eval
+        return eval(f"go.{plot_style}({trace_as_string})")  # pylint: disable=eval-used
```

Comment:
Out of interest did you look into trying something like this? 
```
import sys

def str_to_class(classname):
    return getattr(sys.modules[__name__], classname)
```
Not sure if it will apply here, but use of `eval` is normally avoidable

Reply:
see second answer here:  https://stackoverflow.com/questions/1176136/convert-string-to-python-class-object

Reply:
And you don't get this behaviour when running locally?

Reply:
This behaviour occurs locally and on Travis. I have added a few other assertions to perform other comparisons for the meantime. 
The error should be visible in the last build.
I can run other forms of equality assertion against keys present and the length of both dictionaries.
I just can't seem to run direct equality assertion against both dictionaries using either:
```
self.assertDictEqual(actual, expected)
```
Or
```
self.assertEqual(actual, expected)
```
Further to add to my confusion, it would seem i can run equality checks to see if actual is the same as actual which passes, but i receive the same error present in the last travis build when trying to test equality assertion for expected being the same as expected.  😕 
So:
```self.assertEqual(actual, actual)``` passes
```self.assertEqual(expected, expected)``` fails 

when printing both dictionaries out, they look identical:
```
ACTUAL: 
 {'x': Spectrum
1    100.25
1    100.75
Name: X, dtype: float64, 'y': Spectrum
1    8210
1    6837
Name: Y, dtype: int64, 'error_y': {'type': 'data', 'array': [90.6091, 82.6862], 'visible': False}} 

EXPECTED: 
 {'x': Spectrum
1    100.25
1    100.75
Name: X, dtype: float64, 'y': Spectrum
1    8210
1    6837
Name: Y, dtype: int64, 'error_y': {'type': 'data', 'array': [90.6091, 82.6862], 'visible': False}}
```
Should I consider just removing this comparison assertion for the time being? I feel the other comparison tests added should suffice.